check.jsのプルリクエスト

ShinRookie opened the pull request:
マージお願いします。

Commit:
cehck.jsをコミットします。

Commit:
check.jsのコミット

oriz403 reviewed (commented):
動作確認したものと同様なコードになっていることを確認しました。

naoki0531 commented on src/check.js:
```diff
@@ -0,0 +1,21 @@
+    if ( branchName == 'refs/heads/master' && msg.head_commit.added.length > 0) {
+        console.log('�}�X�^�[�ɑ΂���PUSH�̂��߁A�t�@�C�����擾������J�n���܂��B');
+        fileName = msg.head_commit.added;
+        return fileName;
```
9行目でfileNameを初期化しているため、if文の中でreturnを書く必要がないように思えます
elseに入った場合にreturnをしていないために、後続の処理で判定できないので
return位置の変更をお願いします

tiwuofficial commented on src/check.js:
```diff
@@ -0,0 +1,21 @@
+    const branchName = msg.ref;
+    console.log('msg.ref�̒l��' + branchName);
+    
+    var fileName = '';
```
`var`から`let`に変更をお願いします。

ShinRookie replied:
修正しました。

tiwuofficial commented on src/check.js:
```diff
@@ -0,0 +1,21 @@
+exports.handler = function (event, context) {
```
関数宣言をアロー演算子で書いてください。

ShinRookie replied:
修正しました。

tiwuofficial commented on src/check.js:
```diff
@@ -0,0 +1,21 @@
+    console.log('msg.head_commit.added�̒l��' + msg.head_commit.added);
+    console.log('msg.head_commit.added�̗v�f����' + msg.head_commit.added.length);
+    
+    if ( branchName == 'refs/heads/master' && msg.head_commit.added.length > 0) {
```
`==`ではなく`===`でお願いします。

ShinRookie replied:
修正しました。

tiwuofficial commented on src/check.js:
```diff
@@ -0,0 +1,21 @@
+exports.handler = function (event, context) {
```
全体的に`vent.Records[0].Sns.Message`や`msg.ref`や`msg.head_commit.added`は必ず取得できるものなのでしょうか？
もし取得できないパターンがあるなら存在チェックをお願いします。

ShinRookie replied:
結論から言うと、event.Records[0].Sns.Message、msg.refは必ず取得できます！

git Hub APIの発動タイミングは何かしらの追加、変更、削除のpush後です。
したがって、push件数が0になることはないですし件数は必ず1件になりますので
1件目のpush情報を取ってくる記述である(event.Records[0].Sns.Message)、そのブランチの情報を取ってくる記法である(msg.ref)
で必ず取ってこれます。

head_commit.addedは追加がなく、変更または削除だけの場合は空になります。
こちらは既に動作確認済みなので問題ないです。
head_commit自体は必ず取ってこれます！！

Commit:
index.js、check.jsを以下の内容に修正しました。
●index.js
・getFile.jsの引数をcheck.jsの戻り値に修正

●cehck.js
・ブランチによるマージに対してもreturnするように、returnの場所をif文後に移動
・getfile.jsと変数名を合わせるために、fileNameからfilePathに修正

Commit:
・変数宣言をvarからletにし、ブロックレベルでの変数スコープに変更しました。
・関数宣言をアロー演算子にし、省略記法としました。
・IF文の分岐を「===」にし、値・型完全一致である条件に変更しました。

tiwuofficial commented on src/check.js:
```diff
@@ -0,0 +1,22 @@
+    const msg = JSON.parse(event.Records[0].Sns.Message);
+    console.log(msg);
+    
+    const branchName = msg.ref;
```
こちら変数化の意図はありますでしょうか？

ShinRookie replied:
いまはmsg.refの値はログ出力用で2回、IF分岐用で1回で計3回使用しているので、変数化しました。
2回なら悩みますが、3回使うなら変数化しちゃおうといった感じです。

tiwuofficial commented on src/check.js:
```diff
@@ -0,0 +1,22 @@
+exports.handler = (event, context) => {
```
`exports.handlerで`定義した関数は`const check = require('./src/check'); check(event, context);`では実行時にエラーになります。send.jsがどうやって定義しているか参考にし、修正をお願いします。

ShinRookie replied:
失礼しました。翌営業日までに調べて修正してみます。

ShinRookie replied:
修正しました。

Commit:
関数の宣言部を変更しました。

Commit:
getFile.jsに渡すファイルパスの値を表示するログを追記しました。

naoki0531 commented on src/check.js:
```diff
@@ -0,0 +1,23 @@
+module.exports = (event, context) => {
```
ファイルの文字コードがUTF8ではないようです
ラムダのコンソール上で文字化けしてしまうので、
UTF8に変換をお願いします

ShinRookie replied:
UTF8に変更したものをpushしました

Commit:
エンコードをShift-JisからUTF8に変更しました。
